packages/opentelemetry/src/sampler.ts

```diff
@@ -201,18 +217,28 @@ export function wrapSamplingDecision({
   context,
   spanAttributes,
   sampleRand,
+  sampleRateOverride,
 }: {
   decision: SamplingDecision | undefined;
   context: Context;
   spanAttributes: SpanAttributes;
   sampleRand?: number;
+  sampleRateOverride?: number;
```

Comment:
Maybe this is just the naming, but I am confused. This sounds to me as if we are overriding the used sampler ate for this sampling decision? But if I read the comment below, I guess it means "propagate this sample rate to downstream spans"? Is this a correct interpretation? 😅

Reply:
well here we are in `wrapSamplingDecision`, meaning that the sampling decision was already made. In this context, it kinda means what you wrote: Override the sampling decision on the DSC for everything downstream with this value, no matter what the DSC had as a `sample_rate` before.

Reply:
Renamed it to `shouldUpdateSampleRateOnDownstreamTrace` for now.